e2e/tests/signup/unsuccessfulSignUp.test.js

```diff
@@ -1,89 +1,88 @@
+        await expect(signUpPage.emailAddressErrorMessage).toBeVisible();
     });
 
     it("should type a different password in the confirm field and verify error message", async () => {
         // Registration info
-        await page.fill("#signUpUsername", flow.emailAddress);
-        await page.fill("#signUpPassword", VALID_PASSWORD_USER);
-        await page.fill("#signUpConfirmPassword", CONFIRM_PASSWORD_USER);
-        await page.check("label[for=marketing]");
+        await page.fill(signUpPage.signUpEmailAddressInput, flow.emailAddress);
+        await page.fill(signUpPage.signUpPasswordInput, VALID_PASSWORD_USER);
+        await page.fill(signUpPage.signUpConfirmPasswordInput, CONFIRM_PASSWORD_USER);
+        await page.check(signUpPage.exclusiveTipsAndOffers);
 
-        await page.click("#createAccount_button");
+        // Create the account
+        await page.click(signUpPage.signUpCreateAccountButton);
 
         // Verify error message
-        await expect(page.locator("#errorMessage")).toHaveText("Passwords do not match");
+        await expect(signUpPage.errorMessage).toHaveText(/Passwords do not match/);
 
         // Verify if the appropriate color of borders is displayed
-        const passwordInputParent = page.locator('div.textInput', {has: page.locator('#signUpPassword')});
-        await expect(passwordInputParent).toHaveClass(/textInput__valid/);
-
-        const confirmPasswordInputParent = page.locator('div.textInput', {has: page.locator('#signUpConfirmPassword')});
-        await expect(confirmPasswordInputParent).toHaveClass(/textInput__invalid/);
+        await expect(signUpPage.passwordInputParent).toHaveClass(/textInput__valid/);
+        await expect(signUpPage.confirmPasswordInputParent).toHaveClass(/textInput__invalid/);
     });
 
     it("should attempt to register without checking the checkbox for the terms", async () => {
         // Registration info
-        await page.fill("#signUpUsername", flow.emailAddress);
-        await page.fill("#signUpPassword", VALID_PASSWORD_USER);
-        await page.fill("#signUpConfirmPassword", VALID_PASSWORD_USER);
-        await page.check('label[for="marketing"]');
+        await page.fill(signUpPage.signUpEmailAddressInput, flow.emailAddress);
+        await page.fill(signUpPage.signUpPasswordInput, VALID_PASSWORD_USER);
+        await page.fill(signUpPage.signUpConfirmPasswordInput, VALID_PASSWORD_USER);
+        await page.check(signUpPage.exclusiveTipsAndOffers);
 
-        // Save
-        await page.click("#createAccount_button");
+        // Create the account
+        await page.click(signUpPage.signUpCreateAccountButton);
 
         // Verify the warning message
-        await expect(page.locator("#tAndCsErrorMessage")).toHaveText("Please accept the T&Cs and Privacy Policy");
-    });
+        await expect(signUpPage.tAndCsErrorMessage).toHaveText("Please accept the T&Cs and Privacy Policy");
```

Comment:
please don't leave plain string/text inside the test.

Instead of "Please accept...", add create a directory inside the e2e folder: utils
Inside of it create a new js file called "dataModel.js"
In this file, put all the related text strings used for assertion, and call them inside the test:
dataModel.acceptTCsErrorMsg

Reply:
Ok!